Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1457,8 +1457,12 @@ void HTMLElement::popoverAttributeChanged(const AtomString& value)
         { CSSSelector::PseudoClassClosed, newPopoverState != PopoverState::None }
     });
 
-    if (oldPopoverState != PopoverState::None)
-        hidePopoverInternal(FocusPreviousElement::Yes, FireEvents::No);
+    if (popoverData() && popoverData()->visibilityState() == PopoverVisibilityState::Showing) {
+        auto oldValue = attributeWithoutSynchronization(popoverAttr);
+        hidePopoverInternal(FocusPreviousElement::Yes, FireEvents::Yes);
+        if (!isConnected() || !popoverData() || oldValue != attributeWithoutSynchronization(popoverAttr))
```

Comment:
Checking for the popover showing state addresses the previous case, but the `!isConnected()` check would still break for this case (in fact Chrome has this bug):

```
var div = document.createElement("div");
document.body.append(div);
div.popover = "manual";

div.showPopover();

div.addEventListener("beforetoggle", (e) => {
  if (e.newState == "closed") {
    div.remove();
    setTimeout(() => {
      document.body.append(div);
      div.showPopover();
    }, 0);
  }
}, {once: true});

div.popover = "auto";
```
with this case, the resulting popover has manual behavior despite the attribute value being auto.

I wonder if there are similar bugs with the other two checks.

cc @josepharhar @mfreed7 heads up :)

Reply:
Ohh, yeah that's not good. I'm open to suggestions on how to fix. I'd be supportive of adding a WPT that exposes this bug, also. @josepharhar any thoughts?

Reply:
@mfreed7 The easiest fix is just to remove all these checks and stop firing events, like WebKit already currently does. I guess the "stop firing events" part is up for discussion, but I personally think it's reasonable in this case.

Reply:
(Regarding WPTs, I'm out next week, so I'd appreciate if one of you can write one :) )

Reply:
I filed this HTML spec issue 2 weeks ago fwiw: https://github.com/whatwg/html/issues/9034